src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1753,6 +1753,40 @@ private void deleteRelationIfExists(@Nonnull final SQLiteConnection sqliteConnec
   }
 
   /**
+   * Connects to SQLite to delete
+   * 
+   * @throws CatalogException if there is an error in the catalog.
+   */
+  public void deleteRelationFromCatalog(final RelationKey relationkey) throws CatalogException {
+    if (isClosed) {
+      throw new CatalogException("Catalog is closed.");
+    }
+    /* Do the work */
+    try {
+      queue.execute(new SQLiteJob<Void>() {
+        @Override
+        protected Void job(final SQLiteConnection sqliteConnection) throws CatalogException, SQLiteException {
+          sqliteConnection.exec("BEGIN TRANSACTION;");
+          try {
+            deleteRelationIfExists(sqliteConnection, relationkey);
+            sqliteConnection.exec("COMMIT TRANSACTION;");
+          } catch (SQLiteException e) {
+            try {
+              sqliteConnection.exec("ROLLBACK TRANSACTION;");
+            } catch (SQLiteException e2) {
+              assert true; /* Do nothing */
+            }
+            throw e;
+          }
+          return null;
+        }
+      }).get();
+    } catch (InterruptedException | ExecutionException e) {
```

Comment:
Sure, I'd be happy to talk about it in the meeting too. If we end up doing asynchronous checking and comparing between local and master catalogs, then wouldn't the methods that create relations also need to be changed as well? I like the idea of locking though too. Again, we can bring this up at the meeting.